lib/sensors/src/temperature.rs

```diff
@@ -10,27 +12,37 @@ use hyped_i2c::{HypedI2c, I2cError};
 pub struct Temperature<'a, T: HypedI2c> {
     i2c: &'a mut T,
     device_address: u8,
+    calculate_bounds: fn(f32) -> SensorValueRange<f32>,
 }
 
 impl<'a, T: HypedI2c> Temperature<'a, T> {
     /// Create a new instance of the temperature sensor and attempt to configure it
+    ///
+    /// # Arguments
+    /// * `i2c` - The I2C peripheral to communicate with the temperature sensor
+    /// * `device_address` - The I2C address of the temperature sensor
+    /// * `calculate_bounds` - An optional function to override the default bounds calculation
     pub fn new(
         i2c: &'a mut T,
         device_address: TemperatureAddresses,
+        calculate_bounds: Option<fn(f32) -> SensorValueRange<f32>>,
```

Comment:
i kinda don't like optional args, when would you want to use it?
my guess would be rename this to `new_with_bounds` and you can then have a `new` function that wraps that and passes in the default bounds

Reply:
reason is, constructors that end up being called with `new(a, b, Some(5), None, None, Some(c)` are ugly and hard to read. one pretty solution for big things like that is builders

Reply:
yup good point, I like that better